File: app/src/main/java/net/north101/android/ghplayertracker/data/DeckType.java
package net.north101.android.ghplayertracker.data;

import org.json.JSONException;

public enum DeckType {
    Basic,
    Class,
    Extra;

    public static DeckType parse(String cardDeckData) throws JSONException {
        switch (cardDeckData) {
            case "basic":
                return DeckType.Basic;

            case "class":
                return DeckType.Class;

            case "extra":
                return DeckType.Extra;

            default:
                throw new JSONException(cardDeckData);
        }
    }
}

File: app/src/main/java/net/north101/android/ghplayertracker/data/DeckType.kt
package net.north101.android.ghplayertracker.data

import org.json.JSONException

enum class DeckType {
    Basic,
    Class,
    Extra;

    companion object {
        @Throws(JSONException::class)
        fun parse(cardDeckData: String): DeckType {
            return when (cardDeckData) {
                "basic" -> DeckType.Basic
                "class" -> DeckType.Class
                "extra" -> DeckType.Extra
                else -> throw JSONException(cardDeckData)
            }
        }
    }
}
